src/OpenSolid/Sphere3d.elm

```diff
@@ -0,0 +1,500 @@
+
+@docs Sphere3d
+
+
+# Constants
+
+@docs unit
+
+
+# Constructors
+
+@docs with, around, throughPoints
+
+
+# Properties
+
+@docs centerPoint, radius, diameter, volume, circumference, boundingBox
+
+
+# Queries
+
+@docs contains
+
+
+# Transformations
+
+@docs scaleAbout, rotateAround, translateBy, mirrorAcross
+
+
+# Coordinate conversions
+
+@docs relativeTo, placeIn, projectOnto, projectInto
+
+-}
+
+import OpenSolid.Axis3d as Axis3d exposing (Axis3d)
+import OpenSolid.BoundingBox3d as BoundingBox3d exposing (BoundingBox3d)
+import OpenSolid.Circle2d as Circle2d exposing (Circle2d)
+import OpenSolid.Circle3d as Circle3d exposing (Circle3d)
+import OpenSolid.Direction3d as Direction3d exposing (Direction3d)
+import OpenSolid.Frame3d as Frame3d exposing (Frame3d)
+import OpenSolid.Geometry.Internal as Internal exposing (Sphere3d)
+import OpenSolid.Plane3d as Plane3d exposing (Plane3d)
+import OpenSolid.Point3d as Point3d exposing (Point3d)
+import OpenSolid.Scalar as Scalar
+import OpenSolid.SketchPlane3d as SketchPlane3d exposing (SketchPlane3d)
+import OpenSolid.Vector3d as Vector3d exposing (Vector3d)
+
+
+{--Imports for verifying the examples:
+
+    import Doc.DocTestStatic.Sphere3d exposing (..)
+    import OpenSolid.Axis3d as Axis3d
+    import OpenSolid.BoundingBox3d as BoundingBox3d
+    import OpenSolid.Circle2d as Circle2d
+    import OpenSolid.Circle3d as Circle3d
+    import OpenSolid.Direction3d as Direction3d
+    import OpenSolid.Frame3d as Frame3d exposing (Frame3d)
+    import OpenSolid.Plane3d as Plane3d
+    import OpenSolid.Point2d as Point2d
+    import OpenSolid.Point3d as Point3d exposing (Point3d)
+    import OpenSolid.Scalar as Scalar
+    import OpenSolid.SketchPlane3d as SketchPlane3d
+    import OpenSolid.Sphere3d as Sphere3d
+    import OpenSolid.Vector3d as Vector3d exposing (Vector3d)
+-}
+
+
+{-| -}
+type alias Sphere3d =
+    Internal.Sphere3d
+
+
+{-| The unit sphere, centered on the origin with a radius of 1.
+
+    Sphere3d.unit
+    --> Sphere3d.with
+    -->     { centerPoint = Point3d.origin
+    -->     , radius = 1
+    -->     }
+
+-}
+unit : Sphere3d
+unit =
+    with { centerPoint = Point3d.origin, radius = 1 }
+
+
+{-| Construct a sphere from its center point and radius:
+
+    exampleSphere =
+        Sphere3d.with
+            { centerPoint =
+                Point3d.fromCoordinates ( 1, 2, 1 )
+            , radius = 3
+            }
+
+-}
+with : { centerPoint : Point3d, radius : Float } -> Sphere3d
+with properties =
+    Internal.Sphere3d { properties | radius = abs properties.radius }
+
+
+{-| Construct a sphere around the given axis that passes through the given point.
+
+    point : Point3d
+    point =
+        Point3d.fromCoordinates ( 3, 0, 2 )
+
+    Sphere3d.around Axis3d.z point
+    --> Sphere3d.with
+    -->     { centerPoint =
+    -->         Point3d.fromCoordinates ( 0, 0, 2 )
+    -->     , radius = 3
+    -->     }
+
+-}
+around : Axis3d -> Point3d -> Sphere3d
```

Comment:
I'm a bit hesitant about including `around` since for any given axis and point there are an infinite number of spheres that that are 'around' the given axis (have their center point on the axis) and through the given point (as opposed to circles, where there's only one solution).

I might be tempted to have

```elm
fromGreatCircle : Circle3d -> Sphere3d
fromGreatCircle circle =
    with
        { centerPoint = Circle3d.centerPoint circle
        , radius = Circle3d.radius circle
        }
```

instead (which you could chain with `Circle3d.around` to get the current `Sphere3d.around` behavior), or just leave it out for now and see what use cases come up that need something like this.

Reply:
I removed it for now.